components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/dao/impl/OrganizationManagementDAOImpl.java

```diff
@@ -1534,4 +1525,212 @@ private boolean isDateTimeAndMSSql(String attributeName) throws OrganizationMana
+                                               String applicationAudience, FilterQueryBuilder filterQueryBuilder,
+                                               FilterQueryBuilder parentIdFilterQueryBuilder, String userID)
+            throws OrganizationManagementServerException {
+
+        String sqlStmt = prepareSqlStatement(authorizedSubOrgsOnly, applicationAudience);
+        String getOrgSqlStmtTail = getOrgSqlStmtTail(authorizedSubOrgsOnly, applicationAudience);
+        if (filterQueryBuilder.getHasSubAttribute()) {
+            sqlStmt = sqlStmt.replace("WHERE", INNER_JOIN_UM_ORG_ATTRIBUTE);
+        }
+        sqlStmt = appendFilterQueries(sqlStmt, filterQueryBuilder, parentIdFilterQueryBuilder, getOrgSqlStmtTail,
+                recursive, sortOrder);
+
+        String username = getAuthenticatedUsername();
+        if (StringUtils.isNotEmpty(username)) {
+            username = UserCoreUtil.removeDomainFromName(username);
+        }
+
+        /* The shared user parent user might be created with user ID if there is business user with same name
+        in the child organization. */
+        return sqlStmt.replace(USER_NAME_LIST_PLACEHOLDER, Stream.of(username, userID)
+                .map(name -> "'" + name + "'").collect(Collectors.joining(",")));
+    }
+
+    private String prepareSqlStatement(boolean authorizedSubOrgsOnly, String applicationAudience) {
+
+        if (Boolean.FALSE.equals(CarbonConstants.ENABLE_LEGACY_AUTHZ_RUNTIME)) {
+            return prepareSqlStatementForModernRuntime(authorizedSubOrgsOnly, applicationAudience);
+        } else {
+            return prepareSqlStatementForLegacyRuntime(authorizedSubOrgsOnly);
+        }
+    }
+
+    private String prepareSqlStatementForModernRuntime(boolean authorizedSubOrgsOnly, String applicationAudience) {
+
+        if (authorizedSubOrgsOnly) {
+            if (StringUtils.isNotBlank(applicationAudience)) {
+                return GET_ORGANIZATIONS_WITH_USER_ROLE_ASSOCIATIONS;
+            } else {
+                return GET_ORGANIZATIONS_WITH_USER_ASSOCIATIONS;
+            }
+        } else {
+            return GET_ORGANIZATIONS;
+        }
+    }
+
+    private String prepareSqlStatementForLegacyRuntime(boolean authorizedSubOrgsOnly) {
+
+        if (authorizedSubOrgsOnly) {
+            return GET_ORGANIZATIONS_LEGACY;
+        } else {
+            return GET_ORGANIZATIONS_WITHOUT_PERMISSION_CHECK;
+        }
+    }
+
+    private String getOrgSqlStmtTail(boolean authorizedSubOrgsOnly, String applicationAudience)
+            throws OrganizationManagementServerException {
+
+        if (Boolean.FALSE.equals(CarbonConstants.ENABLE_LEGACY_AUTHZ_RUNTIME)) {
+            return getOrgSqlStmtTailForModernRuntime(authorizedSubOrgsOnly, applicationAudience);
+        } else {
+            return getOrgSqlStmtTailForLegacyRuntime(authorizedSubOrgsOnly);
+        }
```

Comment:
```suggestion
        if (Boolean.FALSE.equals(CarbonConstants.ENABLE_LEGACY_AUTHZ_RUNTIME)) {
            return getOrgSqlStmtTailForModernRuntime(authorizedSubOrgsOnly, applicationAudience);
        }
        return getOrgSqlStmtTailForLegacyRuntime(authorizedSubOrgsOnly);
```

Reply:
This if-else is now removed along with the method call for `getOrgSqlStmtTailForLegacyRuntime()`. Only the logic of `getOrgSqlStmtTailForModernRuntime()` remains.